plugins/outputs/nats/nats_test.go

```diff
@@ -263,3 +268,106 @@ func createStream(t *testing.T, server string, cfg *nats.StreamConfig) {
 	_, err = js.AddStream(cfg)
 	require.NoError(t, err)
 }
+
+func TestWriteWithLayoutIntegration(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+	natsServicePort := "4222"
+
+	container := testutil.Container{
+		Image:        "nats:latest",
+		ExposedPorts: []string{natsServicePort},
+		Cmd:          []string{"--js"},
+		WaitingFor:   wait.ForListeningPort(nat.Port(natsServicePort)),
+	}
+
+	natsInstance := &NATS{
+		Name: "telegraf",
+		Jetstream: &StreamConfig{
+			Name:     "my-telegraf-stream",
+			Subjects: []string{"my-subject.>"},
+		},
+		serializer: &influx.Serializer{},
+		Log:        testutil.Logger{},
+	}
+
+	streamConfigCompareFunc := func(t *testing.T, si *jetstream.StreamInfo) {
+		require.Equal(t, "my-telegraf-stream", si.Config.Name)
+		require.Equal(t, []string{"my-subject.>"}, si.Config.Subjects)
+	}
```

Comment:
Don't you think it's overkill to declare a function for two lines only used in one place? IMO the code will be much more readable if this function is merged into the caller part.

Reply:
I was just coping previous setups from nats_tests.go, I have removed the function.